crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -150,6 +150,24 @@ def compute_resources_available?
       true
     end
 
+    # Check for presence of HA setup, which is a requirement for non-disruptive upgrade
+    def ha_deployed?
+      unless addon_installed? "ha"
+        errors.add(:base, I18n.t("api.crowbar.ha_not_installed"))
```

Comment:
as discussed on IRC, the errors hash is usually used outside the model where you call `@instance.errors` after some action. maybe we could use `self.errors` in the model though I don't know if that is a good practice, but I think here if we return a hash instead of a bool, it should be sufficient.